addons/website/models/website.py

```diff
@@ -968,12 +968,15 @@ def _get_cached(self, field):
         return self._get_cached_values()[field]
 
     @tools.ormcache_context('key', keys=('lang', 'website_id',))
-    def get_customize_show_views(self, key):
+    def get_customize_show_views(self, key, raise_if_not_found=True):
         """ Return the main infos of the customize_show view from the cache """
-        view = self.viewref(key)
-        if not view.customize_show:
+        view = self.viewref(key, raise_if_not_found=raise_if_not_found)
+        if view and not view.customize_show and raise_if_not_found:
             raise ValueError('View %s is not a customize_show view.' % key)
-        return {'id': view.id, 'name': view.name, 'active': view.active}
+        return view and {'id': view.id, 'name': view.name, 'active': view.active} or {}
+
+    def is_enabled(self, key, raise_if_not_found=False):
```

Comment:
is_view_active_or_doesnt_exist or is_view_active_and_raise_if_not_found

Reply:
is_view_active_and_raise_if_not_found but we override in qweb context with a partial

Reply:
bof, on perd l'interet de la cache a ne pas utiliser le même raise_if_not_found entre backend et frontend par défaut.
Car raise_if_not_found doit etre ajouté comme clé! pour éviter que si premier call avec raise=False, le second avec raise=True ne va pas raiser alors qu'il devrait :/

donc raise=true par defaut pour tout le monde ? et on ne le force pas avec un partial, ainsi le enduser peut tjrs mettre lui meme raise=False s'il veut dans son template. Ok pour toi?


```diff
--- addons/website/models/ir_ui_view.py
+++ addons/website/models/ir_ui_view.py
@@ -448,7 +448,7 @@ class View(models.Model):
                 self._context.copy(),
                 main_object=self,
                 website=request.website,
-                is_view_active=request.website.is_enabled,
+                is_view_active= partial(request.website.is_view_active, raise_if_not_found=False),
                 url_for=url_for,
                 res_company=request.website.company_id.sudo(),
                 languages=request.env['res.lang'].get_available(),
diff --git addons/website/models/website.py addons/website/models/website.py
index 85d367f5084..fed2db053c7 100644
--- addons/website/models/website.py
+++ addons/website/models/website.py
@@ -967,16 +967,9 @@ class Website(models.Model):
     def _get_cached(self, field):
         return self._get_cached_values()[field]
 
-    @tools.ormcache_context('key', keys=('lang', 'website_id',))
-    def get_customize_show_views(self, key, raise_if_not_found=True):
-        """ Return the main infos of the customize_show view from the cache """
-        view = self.viewref(key, raise_if_not_found=raise_if_not_found)
-        if view and not view.customize_show and raise_if_not_found:
-            raise ValueError('View %s is not a customize_show view.' % key)
-        return view and {'id': view.id, 'name': view.name, 'active': view.active} or {}
-
-    def is_enabled(self, key, raise_if_not_found=False):
-        return self.get_customize_show_views(key, raise_if_not_found=raise_if_not_found).get('active')
+    @tools.ormcache_context('key', 'raise_if_not_found', keys=('website_id',))
+    def is_view_active(self, key, raise_if_not_found=True):
+        return self.viewref(key, raise_if_not_found=raise_if_not_found).active
```